snapcraft/internal/remote_build/_launchpad.py

```diff
@@ -179,176 +210,115 @@ def create_snap(self, force=False) -> Entry:
             **optional_kwargs,
         )
 
-    def delete_snap(self) -> None:
-        """Remove a snap recipe and all associated files."""
-        try:
-            # snaps.getByName raises NotFound if snap not does not exist.
-            lp_snap = self._lp.snaps.getByName(name=self._lp_name, owner=self._lp_owner)
-        except restfulclient.errors.NotFound:
-            return
-
-        lp_snap.lp_delete()
-
-    def cleanup(self) -> None:
-        """Delete snap and git repository from launchpad."""
-        self.delete_snap()
-        self.delete_git_repository()
+    def _lp_post_new_git_repository(self) -> Entry:
+        return self._lp.git_repositories.new(
+            name=self._lp_name, owner=self._lp_owner, target=self._lp_owner
+        )
 
-    def start_build(self, timeout: int = 5, attempts: int = 5) -> int:
+    def _lp_post_build_request(self, snap: Entry) -> Entry:
         """Initiate a new snap build."""
-        snap = self.create_snap(force=True)
         dist = self._lp.distributions["ubuntu"]
-
-        snap_build_request = snap.requestBuilds(
+        return snap.requestBuilds(
             archive=dist.main_archive,
             channels={
                 "core18": self._core18_channel,
                 "snapcraft": self._snapcraft_channel,
             },
             pocket="Updates",
         )
-        logger.debug("Request URL: {}".format(snap_build_request))
 
-        while attempts > 0 and snap_build_request.status == "Pending":
+    def _create_git_repository(self, force=False) -> Entry:
+        """Create git repository."""
+        if force:
+            self._delete_git_repository()
+
+        logger.debug("Creating source repository on Launchpad...")
+        logger.debug(
+            f"name={self._lp_name} owner={self._lp_owner} target={self._lp_owner}"
+        )
+        return self._lp_post_new_git_repository()
```

Comment:
this name doesn't go easy create ends up posting a new repo?

Reply:
Fixed.